elastic/kibana
# [Infra] Hosts view list only apm and system integration hosts

closes https://github.com/elastic/kibana/issues/188756

## Summary
 
This PR changes how the Hosts View works by focusing only on documents from the system integration and APM. Hosts from other integrations won't show up anymore, even if they have a `host.name` field. Before, we were mainly considering any documents that had a `host.name` field.

We have to keep APM docs because the UI lets users filter by Service Name, and trying to filter them out messed with the functionality.

So, with this update, the Hosts View will still show APM hosts that add noise, but it's necessary to keep things working as expected.

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/9f104509-15f8-45cc-8ee9-e991870a62dc">

Count query
```
GET remote_cluster:metrics*/_search
{
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
          "range": {
            "@timestamp": {
              "gte": "now-15m",
              "lte": "now"
            }
          }
        }
      ],
      "should": [
        {
          "bool": {
            "should": [
              {
                "term": {
                  "event.module": "system"
                }
              },
              {
                "term": {
                  "metricset.module": "system"
                }
              }
            ],
            "minimum_should_match": 1
          }
        },
        {
          "terms": {
            "metricset.name": [
              "transaction"
            ]
          }
        }
      ]
    }
  },
  "aggs": {
    "count": {
      "cardinality": {
        "field": "host.name"
      }
    }
  }
}
```


The `infra/metrics/host` will now return a new `monitored` property, which allows us to identify APM hosts that are not monitored by the system integration and prompt users to onboard data for them and/or review any configuration that might be wrong

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/c5e36cdb-7e2e-45b8-a91a-4c46751256e3">

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/ece1a16e-cf53-4a4f-b5d7-7a429733cb51">


## Performance

I ran the endpoint a few times with a Kibana instance pointing to an oblt cluster, with a 30-day range search

| Before  | After |
| ------------- | ------------- |
| <img width="500px" alt="image" src="https://github.com/user-attachments/assets/2794ff1c-0af6-402e-8432-b1a06233ba9b"> | <img width="500px" alt="image" src="https://github.com/user-attachments/assets/f0cebea8-87aa-4b19-bef4-56947675d23e"> |


The new version performs on average consistently better

<img width="500px" alt="image" src="https://github.com/user-attachments/assets/8de64ad0-12ef-43d5-ba53-ee09124441f7">

## How to test

- Start a local Kibana and ES instance
- Run  `node scripts/synthtrace infra_hosts_with_apm_hosts --scenarioOpts.numInstances=500 --live --clean`
- Navigate to Infrastructure > Hosts
- Filter by Service Name (filter by `service.environment` in the KQL bar too)
- Check the hosts returned and host count
```
